networkmanager/src/main/java/com/dezen/riccardo/networkmanager/NetworkDictionary.java

```diff
@@ -0,0 +1,318 @@
+package com.dezen.riccardo.networkmanager;
+
+import android.content.Context;
+
+import androidx.room.Room;
+
+import com.dezen.riccardo.networkmanager.database_dictionary.PeerDatabase;
+import com.dezen.riccardo.networkmanager.database_dictionary.PeerEntity;
+import com.dezen.riccardo.networkmanager.database_dictionary.ResourceDatabase;
+import com.dezen.riccardo.networkmanager.database_dictionary.ResourceEntity;
+import com.dezen.riccardo.smshandler.SMSPeer;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static com.dezen.riccardo.smshandler.SmsHandler.SMS_HANDLER_LOCAL_DATABASE;
+
+/**
+ * Class implementing Dictionary. Conceived as a double dictionary on SMSPeer and StringResource.
+ * Due to trouble with testing android class. Maps have been used.
+ * @author Riccardo De Zen, Giorgia Bortoletti
+ */
+public class NetworkDictionary implements Dictionary<SMSPeer, StringResource> {
+    private Map<String, String>  peers;
+    private Map<String, String> resources;
+    private NetworkDictionaryDatabase database;
+
+    public NetworkDictionary(Context context){
+        database = new NetworkDictionaryDatabase(context);
+        peers = new HashMap<>();
+        resources = new HashMap<>();
+    }
+
+    //TODO? make this method asynchronous
+    /**
+     * Imports resources and peers from database to variables Map
+     */
+    private void importFromDatabase(){
+        SMSPeer[] smsPeers = database.getPeers();
+        StringResource[] stringResources = database.getResources();
+
+        for(SMSPeer peer : smsPeers)
+            peers.put(peer.getAddress(), "");
+
+        for(StringResource resource : stringResources)
+            resources.put(resource.getName(), resource.getName());
+
+    }
+
+    //TODO? make this method asynchronous
+    /**
+     * Exports resources and peer from maps to database
+     */
+    private void exportToDatabase(){
+
+        for(Map.Entry<String, String> peerEntry : peers.entrySet())
+        {
+            SMSPeer smsPeer = new SMSPeer(peerEntry.getKey());
+            if(!database.containsPeer(smsPeer))
+            {
+                database.addPeer(smsPeer);
+            }
+        }
+
+        for(Map.Entry<String, String> resourceEntry : resources.entrySet())
+        {
+            StringResource stringResource = new StringResource(resourceEntry.getKey(), resourceEntry.getValue());
+            if(!database.containsResource(stringResource)){
+                database.addResource(stringResource);
+            }else{
+                database.updateResource(stringResource);
+            }
+        }
```

Comment:
I don't think it is necessary because it's a simple "for each" and the name of the methods explains what they do. If you read "contains" and then "update" you understand that it updates only if entity is contained. Do you think that it is difficult to understand what they do and I should write some comments?

Reply:
I was able to understand it, but I struggled to understand it.
Also, as the Product Owner said, +1 comment is way better than -1 comment...